h2o-py/h2o/transforms/decomposition.py

```diff
@@ -1,16 +1,21 @@
 from h2o.frame import H2OFrame
-from h2o.utils.typechecks import assert_is_type, Enum
+from h2o.utils.mixin import assign
 from ..estimators.estimator_base import H2OEstimator
+from ..estimators import H2OPrincipalComponentAnalysisEstimator, H2OSingularValueDecompositionEstimator
```

Comment:
I couldn't stand this out-of-sync duplicated logic anymore here (I don't think this is much used besides), so refactored this to use a delegate approach.
This now works as the model obtained from the backend can be resolved on any object, here:
```
self._delegate._model = self
```